src/mbpfan.c

```diff
@@ -211,12 +211,93 @@ t_sensors *retrieve_sensors()
 
 		    s->file = file;
 		    sensors_found++;
+		    if(verbose) {
+		      mbp_log(LOG_INFO, "Found Sensor at %s", path);
+		    }
+
 		}
 
 		free(path);
 		path = NULL;
 	    }
+
 	}
+
+	    // iMac Sensors
+	    // /sys/devices/platform/applesmc.768/
+	    int sensorNo = 0;
+	    char label[80];
+	    char labelType[80] ="";
+	    char labelType2[80] ="";
+	    for(sensorNo = 0; sensorNo<NUM_TEMP_INPUTS; sensorNo++) {
+	      path = smprintf("%s/temp%d_label", APPLESMC_PATH, sensorNo);
+	      FILE *file = fopen(path, "r");
+	      free(path);
+	      if(NULL == file) {
+		continue;
+	      } else {
+		fscanf(file, "%s", label);
+		fclose(file);
+		if (label[1] != 'G' ){ // GPU Only
+		  //		  continue;
```

Comment:
Dead code.

Reply:
You might Try if commenting this in would get it working again with your Macbook?
I think later someone can introduce a Black/Whitelist for sensors.
I one wants to introduce Black/Whitelist it would be nice to use labeel Names for the Configuration. This was another reason to introduce named Labels.

Reply:
Sorry I don't have time to experiment with this.  I recommend narrowing this PR to minimize the chance of regression, even if that requires several PRs and more work for you.